docs/documentation/upgrading/topics/keycloak/changes-24_0_0.adoc

```diff
@@ -284,6 +284,10 @@ Secrets and ConfigMaps referenced via the Keycloak CR will now be polled for cha
 
 This was done so to not require label manipulation on those resources. After upgrading if any Secret still has the operator.keycloak.org/component label, it may be removed or ignored.
 
+= Hostname Verification Policy
+
+The default for `spi-truststore-file-hostname-verification-policy` and the new `tls-hostname-verifier` option is now DEFAULT, rather than WILDCARD. The WILDCARD and STRICT option values have been deprecated - you should simply rely upon DEFAULT instead.
```

Comment:
> It'd be good to elaborate a bit more here, IMHO. As a user I'd like to know about the implications of changing to DEFAULT from WILDCARD and STRICT, respectively.

I've added some more text explaining the differences.

> We should also mention this new behavior in the context of the HTTP client. AFAIK, it's not user configurable directly but the change to DEFAULT still might have some impact I guess? Note that it also affects adapters, AFAIK.

I don't expect that it would. The changes are pretty nuanced / corner case.